test/test-util/src/mz.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::time::Duration;
+
+use tokio_postgres::{Client, Error, NoTls, Row};
+
+/// Create and return a new PostgreSQL client, spawning off the connection
+/// object along the way.
+pub async fn client(user: &str, host: &str, port: u16) -> Result<Client, anyhow::Error> {
+    let (mz_client, conn) = tokio_postgres::Config::new()
+        .user(user)
+        .host(host)
+        .port(port)
+        .connect(NoTls)
+        .await?;
+
+    // The connection object performs the actual communication with the database,
+    // so spawn it off to run on its own.
+    tokio::spawn(async move {
+        if let Err(e) = conn.await {
+            log::info!("connection error: {}", e);
+        }
+    });
+
+    Ok(mz_client)
+}
+
+/// Try running PostgresSQL's `query` function, checking for a common
+/// Materialize error in `check_error`.
+pub async fn try_query(
+    mz_client: &Client,
+    query: &str,
+    delay: Duration,
+) -> Result<Vec<Row>, anyhow::Error> {
+    loop {
+        let timer = std::time::Instant::now();
+        match mz_client.query(&*query, &[]).await {
+            Ok(rows) => return Ok(rows),
+            Err(e) => check_error(e)?,
+        }
+        delay_for(timer.elapsed(), delay).await;
+    }
```

Comment:
Should we use `ore::retry::retry_for` here? Maybe its retry policy isn't configurable enough. Ah well. Food for thought.

Reply:
It's not quite flexible enough! But perhaps this warrants a new retry function

Reply:
Indeed! But def feel free to punt on that.